### PR TITLE
2 implement parsing

### LINKED ISSUE #2: implement parsing

### LINKED ISSUE #14: Need to close fd in read_file

### LINKED ISSUE #19: Texture without .xpm return an invalid color
```
NO ./texture/test
SO ./texture/test
WE ./texture/test
EA ./texture/test
```
print the error 
```
Error
Color's must be compose by three integer, there value must be between 0 and 255
```

### LINKED ISSUE #21: color on wall texture return missing graphical element
```
NO	./texture/test.xpm
SO	./texture/test.xpm
WE	./texture/test.xpm
EA	30,45,82

F	30,45,82
C	30,45,82
```
returns
```
Error
The map doesn't contain all required graphical elements
```

### LINKED ISSUE #22: multiples comas return an valid value
```
F 10,250,0
C ,,,
```
same for `C ,`, `C ,,` and `F 10,250,0,,,,,`
returns an valid map

### PR DESCRIPTION
**Pull request by Neffi42**
- created define  messages in collector and window
- created parse_file.c
- Created usage check
- created texture.h and texture structure
- created vector and maps structures
- created check_suffix and added close statement in read_file
- created texture.c
- use define message rather than harcoded one in error
- renamed texture.h in graphic.h
- updated check_suffix
- renamed texture.c
- updated Makefile
- created t_graphic struc
- created parse_file function
- corrected potential leak if add_collector failed
- put check_suffix in static
- updated graphical structures
- normed
- implemented parsing of color and texture
- put check_rules in static
- created t_pmap structure
- created struct t_player
- added width and heigth to t_map
- created parse_map.c and parse_map function
- updated parse_map and create rectangle of the map
- removed debug print
- normed
- created flood_map
- implemented flood_map

### CODE REVIEW

File: include/cub3d.h [hunk 1]
```diff
@@ -16,7 +16,29 @@
 # include "libft.h"
 # include "mlx.h"
 # include "utils.h"
+# include "graphic.h"
 
 # include <stdio.h>
 
+typedef struct s_vector
+{
+	double	x;
```
Comment (iamantoinev) [inline include/cub3d.h]: May need to chage to float after researches
[explenation here](https://stackoverflow.com/questions/2386772/what-is-the-difference-between-float-and-double)

File: src/read_file.c [hunk 2]
```diff
@@ -15,12 +15,26 @@
...
 #define MALLOC_ERROR "Malloc failed"
 #define OPEN_ERROR "Cannot open file"
+#define SUFFIX_ERROR "Invalid suffix, use a <map.cub>"
 
 void	clear_file(void *v)
 {
 	ft_lstclear((t_list **) &v, &free);
 }
 
+static void	check_suffix(const char *f, t_collector *col)
```
Comment (iamantoinev) [inline src/read_file.c]: Does the file `.cub` should be considered as a valid filename?

File: src/read_file.c [hunk 3]
```diff
@@ -30,21 +44,22 @@ t_list	*read_file(const char *file, t_collector *collector)
 	int		fd;
 
 	lst = NULL;
-	fd = open(file, O_RDONLY);
+	(check_suffix(file, collector), fd = open(file, O_RDONLY));
```
Comment (iamantoinev) [inline src/read_file.c]: May need to refactor function to make this condition explicit

File: src/parse_map.c [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+		map->map[i++] = line->content;
+		line = line->next;
+	}
+}
+
+static int	parse_player(size_t i, size_t j, t_map *map, const char *line)
+{
+	if (map->player.pos.x != 0 && map->player.pos.y != 0)
+		return (1);
+	map->player = (t_player){{i, j}, {0, 0}};
```
Comment (iamantoinev) [inline src/parse_map.c:54]: may create errors if stack allocation is freed

**Review by antoineverin — CHANGES_REQUESTED** [review]
* #22 
* #21 
* #19 
* [issue revelant](https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&pullRequest=18&id=TheSecretOrganization_cub3d&open=AY4dk1HF1r6v5J7mAT73)

File: src/read_file.c (hunk 3 above)
Comment (Neffi42) [reply src/read_file.c]: resolved in #24

File: src/read_file.c (hunk 2 above)
Comment (Neffi42) [reply src/read_file.c]: I guess, I can't see why it shouldn't

File: include/cub3d.h (hunk 1 above)
Comment (Neffi42) [reply include/cub3d.h]: From what I read on the post, it seems that using double is better. It provides much more precision in operations. Why should we use float instead ?

File: src/parse_map.c (hunk 4 above)
Comment (Neffi42) [reply src/parse_map.c:54]: how so ? map's allocation comes from main

File: include/cub3d.h (hunk 1 above)
Comment (iamantoinev) [reply include/cub3d.h]: We don't need that much of precision and float operations are most faster

File: src/read_file.c (hunk 2 above)
Comment (iamantoinev) [reply src/read_file.c]: Ok, just asking to be sure

File: include/cub3d.h (hunk 1 above)
Comment (Neffi42) [reply include/cub3d.h]: it will be done my lord

**Review by antoineverin — APPROVED** [review]

**Pull request merged**